Building with PG 16 fails

with this error:

```
src/byteamagic.c: In function 'magic_bytea_info':
src/byteamagic.c:56:42: warning: implicit declaration of function 'VARDATA_ANY' [-Wimplicit-function-declaration]
   56 |     magic_result = magic_buffer(magic_c, VARDATA_ANY(b),
      |                                          ^~~~~~~~~~~
src/byteamagic.c:57:13: warning: implicit declaration of function 'VARSIZE_ANY_EXHDR' [-Wimplicit-function-declaration]
   57 |             VARSIZE_ANY_EXHDR(b));
      |             ^~~~~~~~~~~~~~~~~
src/byteamagic.c:56:42: warning: passing argument 2 of 'magic_buffer' makes pointer from integer without a cast [-Wint-conversion]
   56 |     magic_result = magic_buffer(magic_c, VARDATA_ANY(b),
      |                                          ^~~~~~~~~~~~~~
      |                                          |
      |                                          int
In file included from src/byteamagic.c:11:
/usr/include/magic.h:134:35: note: expected 'const void *' but argument is of type 'int'
  134 | const char *magic_buffer(magic_t, const void *, size_t);
      |                                   ^~~~~~~~~~~~
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -O2 -fPIC -fvisibility=hidden -shared -o byteamagic.so src/byteamagic.o -L/usr/local/lib   -L/usr/lib/llvm15/lib  -L/usr/local/lib -Wl,--as-needed -fvisibility=hidden -lmagic 
cp sql/byteamagic.sql sql/byteamagic--0.2.2.sql
clang-15 -Wno-ignored-attributes -fno-strict-aliasing -fwrapv -Xclang -no-opaque-pointers -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -O2  -I. -I./ -I/usr/local/include/postgresql/server -I/usr/local/include/postgresql/internal  -D_GNU_SOURCE -I/usr/include/libxml2  -flto=thin -emit-llvm -c -o src/byteamagic.bc src/byteamagic.c
src/byteamagic.c:56:42: warning: call to undeclared function 'VARDATA_ANY'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    magic_result = magic_buffer(magic_c, VARDATA_ANY(b),
                                         ^
src/byteamagic.c:57:13: warning: call to undeclared function 'VARSIZE_ANY_EXHDR'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
            VARSIZE_ANY_EXHDR(b));
            ^
src/byteamagic.c:56:42: error: incompatible integer to pointer conversion passing 'int' to parameter of type 'const void *' [-Wint-conversion]
    magic_result = magic_buffer(magic_c, VARDATA_ANY(b),
                                         ^~~~~~~~~~~~~~
/usr/include/magic.h:134:47: note: passing argument to parameter here
const char *magic_buffer(magic_t, const void *, size_t);
                                              ^
2 warnings and 1 error generated.
make: *** [/usr/local/lib/postgresql/pgxs/src/makefiles/../../src/Makefile.global:1080: src/byteamagic.bc] Error 1
```

It worked fine with PG 15.

--- a/Changes
+++ b/Changes
@@ -1,4 +1,7 @@
 Revision history for PostgreSQL extension explanation.
 
+Unreleased
+      - Support for PG 16+ by including the varatt.h header
+
 0.2.2   2012-05-16T07:28:09
       - Fixed linling error. 

--- a/src/byteamagic.c
+++ b/src/byteamagic.c
@@ -10,6 +10,12 @@
 #include "utils/builtins.h"
 #include "magic.h"
 
+#if PG_MAJORVERSION_NUM >= 16
+// varatt.h was split away from postgres.h in PG 16.
+// https://github.com/postgres/postgres/commit/d952373a987bad331c0e499463159dd142ced1ef
+#include "varatt.h"
+#endif
+
 #ifdef PG_MODULE_MAGIC
 PG_MODULE_MAGIC;
 #endif

